download_and_gen.py

```diff
@@ -229,8 +229,9 @@ def __init__(self, base_url: str, metrics_url: str):
             mapfile = csv.reader(mapfile_csv_lines)
             first_row = True
             for l in mapfile:
-                if len(l) == 6:
-                    l.append("")
+                while len(l) < 7:
```

Comment:
Ian, Sorry about hiccups in the mapfile. Thanks for working around them! I've added a local check for both column count as well as uncore experimental.

Reply:
Thanks! I'm unclear what to do about the atom case with Alderlake CPUID 6-BE, will that change in the future? Perhaps we need to do something special for that CPU in Linux as normally on a hybrid system the PMU for atom is called cpu_atom, but maybe on this model it is cpu_core or just cpu?
Wrt local checks, you may want to check what is checked here that each model maps to a single path (except for hybrid where we expect exactly 2) and that things like core/uncore/experimental uncore agree with the what is in the path. There was previously an issue for this and SPR.
Something I've also raised is that there are fewer CPUIDs here compared to:
https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/intel-family.h#L45
and the perf tool has also listed CPUIDs not in this file previously (A7 and 7D both icelake variants). It looks like palm cove, sunny cove, lightning mountain and quark x1000 are the only missing cases now - likely intentionally :-)

Reply:
Hi Ian, Thanks for the feedback.

- I'm unclear what to do about the atom case with Alderlake CPUID 6-BE, will that change in the future? Perhaps we need to do something special for that CPU in Linux as normally on a hybrid system the PMU for atom is called cpu_atom, but maybe on this model it is cpu_core or just cpu?
  - I believe the current 0xBE mapfile entry is a correct representation of Alder Lake N (https://github.com/torvalds/linux/blob/534b0abc627a4034ecaf177e780b1680e61b37f4/arch/x86/include/asm/intel-family.h#L112) . Sorry about the additional complexity.
- Wrt local checks, you may want to check what is checked here that each model maps to a single path (except for hybrid where we expect exactly 2)
   - Thanks. I'll add a check.
- Things like core/uncore/experimental uncore agree with the what is in the path.
   - Definitely. I have checks for uncore vs uncore experimental since I broke Snow Ridge recently. I'll try to expand these types of checks soon.
- There are fewer CPUIDs here compared to:
https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/intel-family.h#L45
   - This was a great suggestion. As you've noticed the mapfile is updated with only a few missing. The current missing IDs are intentionally not added.

Reply:
Great! Is there anything outstanding from this pull request? Would be nice to land it.